### PR TITLE
Added 2 test cases to handle the name validation for createPlayList

### PR DESCRIPTION
**Pull request by parthiban-samykutti**

### CODE REVIEW

File: src/main/java/com/musicshare/controller/PlaylistController.java [hunk 1]
```diff
@@ -20,13 +20,19 @@ public PlaylistController(PlaylistService service) {
...
-    public PlayListResponse createPlayList(@PathVariable String playListName) {
-
-        PlayListResponse response = PlayListResponse.builder()
-                .message("playlist created successfully.")
-                .data(service.createPlaylist(playListName))
-                .build();
+    @PostMapping
+    public PlayListResponse createPlayList(@RequestParam("playListName") String playListName) {
+        PlayListResponse response = null;
+        if (StringUtils.hasText(playListName)) {
```
Comment (vyomrastogi) [inline src/main/java/com/musicshare/controller/PlaylistController.java]: we can have two url paths defined here with one returning error message

File: src/main/java/com/musicshare/model/PlayListResponse.java [hunk 2]
```diff
@@ -8,5 +10,6 @@
 @Builder
 public class PlayListResponse {
     private String message;
+    @JsonInclude(JsonInclude.Include.NON_NULL)
```
Comment (vyomrastogi) [inline src/main/java/com/musicshare/model/PlayListResponse.java]: we can continue to send it as null

**Review by vyomrastogi — CHANGES_REQUESTED** [review]

File: src/main/java/com/musicshare/controller/PlaylistController.java (hunk 1 above)
Comment (parthiban-samykutti) [reply src/main/java/com/musicshare/controller/PlaylistController.java]: updated.

File: src/main/java/com/musicshare/model/PlayListResponse.java (hunk 2 above)
Comment (parthiban-samykutti) [reply src/main/java/com/musicshare/model/PlayListResponse.java]: done

File: src/test/java/com/musicshare/controller/PlaylistControllerTest.java [hunk 3]
```diff
@@ -29,9 +30,9 @@
     @DisplayName("createPlaylist - with a name")
     public void testCreatePlaylistWithName() throws Exception {
         when(service.createPlaylist(any())).thenReturn(Playlist.builder().name("playList1").id(1l).songs(new ArrayList<>()).build());
-        mockMvc.perform(post("/api/v1/playlist/?playListName={playListName}", "playList1")
-        )
+        mockMvc.perform(post("/api/v1/playlist/{playListName}", "playList1") )
                 .andExpect(status().isOk())
+                .andDo(print())
                 .andExpect(jsonPath("$.data").exists())
                 .andExpect(jsonPath("$.message").value("playlist created successfully."));
```
Comment (vyomrastogi) [inline src/test/java/com/musicshare/controller/PlaylistControllerTest.java]: we can enhance testcase to validate id exists and  name is same as passed

**Review by vyomrastogi — CHANGES_REQUESTED** [review]

**Review by vyomrastogi — APPROVED** [review]

**Comment by vyomrastogi** [comment]
Jenn approved

**Pull request merged**